pkg/resources/aws/aws.go

```diff
@@ -1828,8 +1863,13 @@ func ListRoute53Records(cloud fi.Cloud, clusterName string) ([]*resources.Resour
 				prefix := strings.TrimSuffix(name, clusterName)
 
 				remove := false
-				// TODO: Compute the actual set of names?
-				if prefix == ".api" || prefix == ".api.internal" || prefix == ".bastion" {
+				// TODO: Compute the actual set of names for internal records?
```

Comment:
We can also run without an AWS ELB entirely - that's actually the default mode if you just run `kops create cluster` without `topology=private` etc.  We still need to clean up this record.

Reply:
makes sense, so do you think dynamically checking the value of `masterPublicName` and then deleting would make sense instead of hardcoding the check to `api.internal`?

Reply:
@justinsb I've made changes such that we will verify the public api endpoint with what is available in `spec.masterPublicName` and then delete. Please review the changes.